mailbox/lucene/src/main/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndex.java

```diff
@@ -886,65 +842,57 @@ private Query createHeaderQuery(SearchQuery.HeaderCriterion crit) throws Unsuppo
         String fieldName = PREFIX_HEADER_FIELD + name;
-        if (op instanceof SearchQuery.ContainsOperator) {
-            ContainsOperator cop = (ContainsOperator) op;
-            return createTermQuery(fieldName, cop.getValue().toUpperCase(Locale.US));
-        } else if (op instanceof SearchQuery.ExistsOperator) {
-            return new PrefixQuery(new Term(fieldName, ""));
-        } else if (op instanceof SearchQuery.DateOperator) {
-                DateOperator dop = (DateOperator) op;
+        switch (op) {
+            case ContainsOperator cop -> {
+                return createTermQuery(fieldName, cop.getValue().toUpperCase(Locale.US));
+            }
+            case SearchQuery.ExistsOperator existsOperator -> {
+                return new PrefixQuery(new Term(fieldName, ""));
+            }
+            case DateOperator dop -> {
                 String field = toSentDateField(dop.getDateResultion());
                 return createQuery(field, dop);
-        } else if (op instanceof SearchQuery.AddressOperator) {
-            String field = name.toLowerCase(Locale.US);
-            return createTermQuery(field, ((SearchQuery.AddressOperator) op).getAddress().toUpperCase(Locale.US));
-        } else {
-            // Operator not supported
-            throw new UnsupportedSearchException();
+            }
+            case SearchQuery.AddressOperator addressOperator -> {
+                String field = name.toLowerCase(Locale.US);
+                return createTermQuery(field, addressOperator.getAddress().toUpperCase(Locale.US));
+            }
+            case null, default ->
+                // Operator not supported
+                throw new UnsupportedSearchException();
         }
     }
 
 
-    private Query createQuery(String field, DateOperator dop) throws UnsupportedSearchException {
+    private Query createQuery(String field, DateOperator dop) {
         Date date = dop.getDate();
         DateResolution res = dop.getDateResultion();
         DateTools.Resolution dRes = toResolution(res);
         String value = DateTools.dateToString(date, dRes);
-        switch (dop.getType()) {
-        case ON:
-            return new TermQuery(new Term(field, value));
-        case BEFORE:
-            return TermRangeQuery.newStringRange(field, DateTools.dateToString(MIN_DATE, dRes), value, true, false);
-        case AFTER:
-            return TermRangeQuery.newStringRange(field, value, DateTools.dateToString(MAX_DATE, dRes), false, true);
-        default:
-            throw new UnsupportedSearchException();
-        }
+        return switch (dop.getType()) {
+            case ON -> new TermQuery(new Term(field, value));
+            case BEFORE ->
+                TermRangeQuery.newStringRange(field, DateTools.dateToString(MIN_DATE, dRes), value, true, false);
+            case AFTER ->
+                TermRangeQuery.newStringRange(field, value, DateTools.dateToString(MAX_DATE, dRes), false, true);
```

Comment:
```suggestion
            case BEFORE -> TermRangeQuery.newStringRange(field, DateTools.dateToString(MIN_DATE, dRes), value, true, false);
            case AFTER -> TermRangeQuery.newStringRange(field, value, DateTools.dateToString(MAX_DATE, dRes), false, true);
```

And I really do not like those magic boolean constants...